pageserver/src/metrics.rs

```diff
@@ -525,6 +525,15 @@ static RESIDENT_PHYSICAL_SIZE: Lazy<UIntGaugeVec> = Lazy::new(|| {
     .expect("failed to define a metric")
 });
 
+static VISIBLE_PHYSICAL_SIZE: Lazy<UIntGaugeVec> = Lazy::new(|| {
+    register_uint_gauge_vec!(
+        "pageserver_visible_physical_size",
+        "The size of the layer files present in the pageserver's filesystem.",
+        &["tenant_id", "shard_id", "timeline_id"]
+    )
+    .expect("failed to define a metric")
+});
```

Comment:
Hmm, I think it would be a bit hard to interpret, because this covers two cases:
- download a layer for timeline->lsn lookup.
- download a layer after we create a branch (layers are only marked visible the first time they're accessed in this case)

Reply:
The main metric I'd be looking at for effectiveness is the on-demand downloads: if we see a spike of these after we start using visibility to drive evictione, we know we goofed somehow.

Reply:
> download a layer after we create a branch

It would makes sense to recompute the coverage when creating a branch later on.

> download a layer for timeline->lsn lookup

We could special case this.

> The main metric I'd be looking at for effectiveness is the on-demand downloads: if we see a spike of these after we start using visibility to drive evictione, we know we goofed somehow.

This makes sense though.

Reply:
>It would makes sense to recompute the coverage when creating a branch later on.

Yeah.  For this PR I'm trying to be minimal about calls to the function, with just enough to make the results somewhat stable across restarts etc.